### PR TITLE
versions:bump: handle `backstage:^` versions

### PR DESCRIPTION
**Pull request by mtlewis**
## Hey, I just made a Pull Request!

Introduces basic support for `backstage:^` versions in the versions:bump command in backstage-cli. The behavior of the command changes in the following ways:

1. If the yarn plugin is detected, write `backstage:^` versions in package.json files in the repo where appropriate.
2. Update breaking change detection to correctly resolve lockfile entries for packages using backstage:^ versions.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/cli/src/commands/versions/bump.ts [hunk 1]
```diff
@@ -223,12 +233,27 @@ export default async (opts: OptionValues) => {
...
+              const oldLockfileRange = await asLockfileVersion(oldRange);
+
+              // Don't use backstage:^ versions for peerDependencies; they only
+              // support npm and workspace: versions.
+              const useBackstageRange =
+                hasYarnPlugin && depType !== 'peerDependencies';
+
+              const newRange = useBackstageRange ? 'backstage:^' : dep.range;
+
+              pkgJson[depType][dep.name] = newRange;
```
Comment (Rugvip) [inline packages/cli/src/commands/versions/bump.ts:251]: This seems a bit aggressive, we probably don't want this to happen unless the version manifest we're bumping to also contains the package?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! 👍 

Just a QQ on the `package.json` replacement

File: packages/cli/src/commands/versions/bump.ts (hunk 1 above)
Comment (mtlewis) [reply packages/cli/src/commands/versions/bump.ts:212]: Yep, makes sense! I guess this mostly covers downgrades, and packages moving out of the @backstage scope and into the community repo.

Comment (mtlewis) [reply packages/cli/src/commands/versions/bump.ts:212]: Done in 9def475ed084.

File: packages/cli/src/commands/versions/bump.ts [hunk 2]
```diff
@@ -25,6 +25,8 @@ import chalk from 'chalk';
 import ora from 'ora';
 import semver from 'semver';
 import { OptionValues } from 'commander';
+import yaml from 'js-yaml';
```
Comment (Rugvip) [inline packages/cli/src/commands/versions/bump.ts:28]: Use `yaml` here instead if possible, since that's what we already use in other places

**Review by Rugvip — APPROVED** [review]
Alright, looks good! 👍 

If possible let's switch out the YAML parser, but otherwise keep as is. Pre-approved and fine to ship either way :shipit:

**Pull request merged**